src/DataSourceProvider.vue

```diff
@@ -1,19 +1,138 @@
 <template>
-  <div>
-    <h1>Your fliplet-widget-data-source-provider widget will be added here.</h1>
-    <First-Component></First-Component>
-  </div>
+  <section>
+    <div v-if="isLoading" class="spinner-holder animated">
+      <div class="spinner-overlay">Loading...</div>
+    </div>
+    <div v-else-if="hasError" class="alert alert-danger" role="alert">
+      {{ errorMessage }}
+    </div>
+    <div v-else>
+      <DataSourceSelector
+        :currentAppDataSources="appDataSources" :otherDataSources="allDataSources"
+        :selectedDataSource="selectedDataSource"
+        :changeDataSource="changeDataSource"
+        :showAll="showAll"
```

Comment:
The `showAll` behaviour could just be internal to the `DataSourceSelector` component because it's related to the "select a data source" functionality of the provider.

Reply:
Same as above only for the correct display of the checkbox.